lasagne/layers/recurrent.py

```diff
@@ -0,0 +1,1350 @@
+        else:
+            self.nonlinearity_out = nonlinearity_out
+
+        self.learn_init = learn_init
+        self.num_units = num_units
+        self.backwards = backwards
+        self.peepholes = peepholes
+        self.gradient_steps = gradient_steps
+        self.grad_clipping = grad_clipping
+        self.unroll_scan = unroll_scan
+        self.precompute_input = precompute_input
+
+        if unroll_scan and gradient_steps != -1:
+            raise ValueError(
+                "Gradient steps must be -1 when unroll_scan is true.")
+
+        # Retrieve the dimensionality of the incoming layer
+        input_shape = self.input_shapes[0]
+
+        if unroll_scan and input_shape[1] is None:
+            raise ValueError("Input sequence length cannot be specified as "
+                             "None when unroll_scan is True")
+
+        num_inputs = np.prod(input_shape[2:])
+
+        def add_gate_params(gate, gate_name):
+            """ Convenience function for adding layer parameters from a Gate
+            instance. """
+            return (self.add_param(gate.W_in, (num_inputs, num_units),
+                                   name="W_in_to_{}".format(gate_name)),
+                    self.add_param(gate.W_hid, (num_units, num_units),
+                                   name="W_hid_to_{}".format(gate_name)),
+                    self.add_param(gate.b, (num_units,),
+                                   name="b_{}".format(gate_name),
+                                   regularizable=False),
+                    gate.nonlinearity)
+
+        # Add in parameters from the supplied Gate instances
+        (self.W_in_to_ingate, self.W_hid_to_ingate, self.b_ingate,
+         self.nonlinearity_ingate) = add_gate_params(ingate, 'ingate')
+
+        (self.W_in_to_forgetgate, self.W_hid_to_forgetgate, self.b_forgetgate,
+         self.nonlinearity_forgetgate) = add_gate_params(forgetgate,
+                                                         'forgetgate')
+
+        (self.W_in_to_cell, self.W_hid_to_cell, self.b_cell,
+         self.nonlinearity_cell) = add_gate_params(cell, 'cell')
+
+        (self.W_in_to_outgate, self.W_hid_to_outgate, self.b_outgate,
+         self.nonlinearity_outgate) = add_gate_params(outgate, 'outgate')
+
+        # Stack input weight matrices into a (num_inputs, 4*num_units)
+        # matrix, which speeds up computation
+        self.W_in_stacked = T.concatenate(
+            [self.W_in_to_ingate, self.W_in_to_forgetgate,
+             self.W_in_to_cell, self.W_in_to_outgate], axis=1)
+
+        # Same for hidden weight matrices
+        self.W_hid_stacked = T.concatenate(
+            [self.W_hid_to_ingate, self.W_hid_to_forgetgate,
+             self.W_hid_to_cell, self.W_hid_to_outgate], axis=1)
+
+        # Stack biases into a (4*num_units) vector
+        self.b_stacked = T.concatenate(
+            [self.b_ingate, self.b_forgetgate,
+             self.b_cell, self.b_outgate], axis=0)
```

Comment:
> I wouldn't be so sure there isn't a use-case for them as attributes

Maybe. But I assume they're just here because one of the earlier implementations actually stacked the numpy arrays before turning them into a shared variable, and this had to happen in the constructor. Let's see if @skaee has any objections against moving that code.